tests/integration/test_charm.py

```diff
@@ -71,30 +33,37 @@ async def test_prometheus_scrape_relation_with_prometheus_tester(
     prometheus_app_name = "prometheus"
     tester_app_name = "prometheus-tester"
 
-    await ops_test.model.deploy(
-        prometheus_charm, resources=prometheus_resources, application_name=prometheus_app_name
-    )
-    await ops_test.model.deploy(
-        prometheus_tester_charm, resources=tester_resources, application_name=tester_app_name
-    )
-    await ops_test.model.wait_for_idle(apps=[prometheus_app_name], status="active")
-    await ops_test.model.wait_for_idle(apps=[tester_app_name], status="active")
-    await ops_test.model.block_until(
-        lambda: len(ops_test.model.applications[prometheus_app_name].units) > 0
+    app_names = [prometheus_app_name, tester_app_name]
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            prometheus_charm, resources=prometheus_resources, application_name=prometheus_app_name
+        ),
+        ops_test.model.deploy(
+            prometheus_tester_charm, resources=tester_resources, application_name=tester_app_name
+        ),
     )
+
+    await ops_test.model.wait_for_idle(apps=app_names, status="active")
+
     await ops_test.model.block_until(
-        lambda: len(ops_test.model.applications[tester_app_name].units) > 0
+        lambda: (
+            len(ops_test.model.applications[prometheus_app_name].units) > 0
+            and len(ops_test.model.applications[tester_app_name].units) > 0
+        )
     )
 
-    assert ops_test.model.applications[prometheus_app_name].units[0].workload_status == "active"
-    assert ops_test.model.applications[tester_app_name].units[0].workload_status == "active"
-
+    assert initial_workload_is_ready(ops_test, app_names)
```

Comment:
This seems to be redundant because `wait_for_idle` is doing exactly that.

Reply:
This is not completely true, as `wait_for_idle` also resolves for unitless apps that stay idle for long enough.